Update /godmode

bartico6 opened the pull request:
- Previous Bouncer checks for GodMode (namely, Hurt) were removed.
- The command now uses the GodmodePower from core Terraria
- The toggle powers (which this command will now make use of) are now
  reset on disconnect to prevent accidentally "gifting" godmode to an
  unsuspecting player.

I'm the original copyright owner for the code in this commit (it is a backport from Frostspark.Server, which I am the sole author of, as mentioned in #2209)

<!-- Warning: If you create a pull request and wish to remain anonymous, you are highly advised to use Tails (https://tails.boum.org/) or a fresh git environment. We will *not* be able to help with anonymization after your pull request has been created. -->

bartico6 committed:
Update /godmode to use JourneyMode Godmode power.

- Previous Bouncer checks for GodMode (namely, Hurt) were removed.
- The command now uses the GodmodePower from core Terraria
- The toggle powers (which this command will now make use of) are now
  reset on disconnect to prevent accidentally "gifting" godmode to an
  unsuspecting player.

bartico6 committed:
Update changelog

QuiCM commented on TShockAPI/Commands.cs:
```diff
@@ -6445,6 +6447,10 @@ private static void ToggleGodMode(CommandArgs args)
 
 			playerToGod.GodMode = !playerToGod.GodMode;
 
+			var god_power = CreativePowerManager.Instance.GetPower<GodmodePower>();
```
Minor nitpick, could this be adjusted to `godPower` to keep style consistency?

bartico6 replied:
Yes, done & marking as resolved :)

bartico6 committed:
Update casing & remove using static

- god_power -> godPower
- Remove using static on GameContent.CreativePowers, instead fully
  qualify the nested class.

QuiCM reviewed (dismissed):
LGTM

hakusaro reviewed (changes requested):
Please warn players that they need to disable god mode before disconnecting as part of the grant message when a user enables god mode. Also, please revise the message when a user joins without permission for god mode, and tell them that they had it on, and we turned it off because they didn't have permission.

bartico6 committed:
Address JourneyMode Godmode issues

- Add /ungodme to allow unstucking godmode for involuntarily godmodded
  characters
- Warn player about disabling Godmode before disconnecting
- Minor change to command format to reduce code copypaste.

hakusaro reviewed (dismissed):
![lgtm_trick](https://user-images.githubusercontent.com/532647/114994468-a6a06080-9e51-11eb-88ac-e533552413d2.gif)

bartico6 committed:
Added feedback to /ungodme

- The command now tells the user that the operation succeeded.
  Previously it would be a silent command, potentially leaving the user
  wondering if it worked

hakusaro reviewed (approved):
![lgtm_trick](https://user-images.githubusercontent.com/532647/114994673-de0f0d00-9e51-11eb-85cf-33d7af6045dc.gif)
